snappy/snap_remote_repo.go

```diff
@@ -220,7 +219,7 @@ func (s *SnapUbuntuStoreRepository) Details(name string, origin string) (parts [
 	case resp.StatusCode == 404:
 		return nil, ErrPackageNotFound
 	case resp.StatusCode != 200:
-		return parts, fmt.Errorf("SnapUbuntuStoreRepository: unexpected http statusCode %v for %s", resp.StatusCode, snapName)
+		return nil, fmt.Errorf("SnapUbuntuStoreRepository: unexpected http statusCode %v for %s", resp.StatusCode, snapName)
```

Comment:
This is oriented to the user, so we shouldn't mention code parts. Perhaps something like

`"unexpected HTTP status code %d while looking for snap %q"

?